sdk/src/wallet/utxo.ts

```diff
@@ -279,33 +310,51 @@ export async function estimateTxFee(
     // TODO: allow submitting the UTXOs, fee estimate and confirmed transactions
     // to avoid fetching them again.
     const esploraClient = new EsploraClient(addressInfo.network);
+    const ordinalsClient = new OrdinalsClient(addressInfo.network);
 
     let confirmedUtxos: UTXO[] = [];
+    let outputsFromAddress: OutputJson[] = [];
+
     if (feeRate) {
-        confirmedUtxos = await esploraClient.getAddressUtxos(fromAddress);
+        [confirmedUtxos, outputsFromAddress] = await Promise.all([
+            esploraClient.getAddressUtxos(fromAddress),
+            // cardinal = return UTXOs not containing inscriptions or runes
+            addressInfo.type === AddressType.p2tr ? ordinalsClient.getOutputsFromAddress(fromAddress, 'cardinal') : [],
+        ]);
     } else {
-        [confirmedUtxos, feeRate] = await Promise.all([
+        [confirmedUtxos, feeRate, outputsFromAddress] = await Promise.all([
             esploraClient.getAddressUtxos(fromAddress),
             esploraClient.getFeeEstimate(confirmationTarget),
+            // cardinal = return UTXOs not containing inscriptions or runes
+            addressInfo.type === AddressType.p2tr ? ordinalsClient.getOutputsFromAddress(fromAddress, 'cardinal') : [],
         ]);
     }
 
     if (confirmedUtxos.length === 0) {
         throw new Error('No confirmed UTXOs');
     }
 
-    const possibleInputs = await Promise.all(
+    const outpointsSet = new Set(outputsFromAddress.map((output) => output.outpoint));
+    const possibleInputs: Input[] = [];
+
+    await Promise.all(
         confirmedUtxos.map(async (utxo) => {
             const hex = await esploraClient.getTransactionHex(utxo.txid);
             const transaction = Transaction.fromRaw(Buffer.from(hex, 'hex'), { allowUnknownOutputs: true });
-
-            return getInputFromUtxoAndTx(
+            const input = getInputFromUtxoAndTx(
                 addressInfo.network as BitcoinNetworkName,
                 utxo,
                 transaction,
                 addressInfo.type,
                 publicKey
             );
+
+            // to support taproot addresses we want to exclude outputs which contain inscriptions
+            if (addressInfo.type === AddressType.p2tr) {
+                if (outpointsSet.has(OutPoint.toString(utxo))) possibleInputs.push(input);
+            } else {
+                possibleInputs.push(input);
+            }
```

Comment:
Minor nit / suggested change:
```suggestion
            if (addressInfo.type === AddressType.p2tr && outpointsSet.has(OutPoint.toString(utxo))) {
                possibleInputs.push(input);
            } else {
                possibleInputs.push(input);
            }
```

Reply:
This is basically a different condition.
Let's say address type is `p2tr` then `input` will be treated as `possible input` regardless whether it's a part `outpointsSet` or not.

Reply:
Ah no you're right nevermind